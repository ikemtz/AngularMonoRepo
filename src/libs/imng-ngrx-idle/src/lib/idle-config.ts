import { InjectionToken } from '@angular/core';

export interface IdleConfig {
  timeoutWarningInMs: number;
  autoLogoutInMs: number;
}
export const IDLE_CONFIG = new InjectionToken<IdleConfig>('idle-config');
