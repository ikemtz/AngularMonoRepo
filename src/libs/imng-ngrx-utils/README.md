# imng-ngrx-utils

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test imng-ngrx-utils` to execute the unit tests.
