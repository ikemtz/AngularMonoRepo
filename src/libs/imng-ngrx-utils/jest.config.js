module.exports = {
  name: 'imng-ngrx-utils',
  preset: '../../jest.config.js',
  coverageDirectory: '../../coverage/libs/imng-ngrx-utils',
  snapshotSerializers: [
    'jest-preset-angular/build/AngularNoNgAttributesSnapshotSerializer.js',
    'jest-preset-angular/build/AngularSnapshotSerializer.js',
    'jest-preset-angular/build/HTMLCommentSerializer.js',
  ],
};
