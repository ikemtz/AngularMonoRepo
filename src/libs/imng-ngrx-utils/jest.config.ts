import type { Config } from 'jest';

const config: Config = {
  displayName: 'imng-ngrx-utils',
  reporters: [
    'default',
    [
      'jest-junit',
      {
        outputDirectory: '.',
        outputName: './junit.imng-ngrx-utils.xml',
      },
    ],
  ],
  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
  globals: {},
  coverageDirectory: '../../coverage/libs/imng-ngrx-utils',
  transform: {
    '^.+\\.(ts|mjs|js|html)$': [
      'jest-preset-angular',
      {
        tsconfig: '<rootDir>/tsconfig.spec.json',
        stringifyContentPathRegex: '\\.(html|svg)$',
      },
    ],
  },
  transformIgnorePatterns: ['node_modules/(?!.*\\.mjs$)'],
  snapshotSerializers: [
    'jest-preset-angular/build/serializers/no-ng-attributes',
    'jest-preset-angular/build/serializers/ng-snapshot',
    'jest-preset-angular/build/serializers/html-comment',
  ],
  preset: '../../jest.preset.js',
};

export default config;
