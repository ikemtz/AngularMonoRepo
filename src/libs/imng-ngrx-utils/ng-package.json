{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/libs/imng-ngrx-utils",
  "lib": {
    "entryFile": "src/lib/public_api.ts"
  },
  "deleteDestPath": true
}
