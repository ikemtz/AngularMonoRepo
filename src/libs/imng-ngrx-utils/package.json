{
  "name": "imng-ngrx-utils",
  "version": "0.0.4",
  "author": "Isaac Martinez <@ikemtz>",
  "license": "MIT",
  "peerDependencies": {
    "@angular/common": "^8.0.0",
    "@angular/core": "^8.0.0",
    "@progress/kendo-data-query": "^1.5.1",
    "@ngrx/store": "^8.0.1"
  },
  "devDependencies": {},
  "ngPackage": {
    "$schema": "./node_modules/ng-packagr/ng-package.schema.json",
    "lib": {
      "entryFile": "src/index.ts"
    },
    "dest": "../../dist/libs/imng-ngrx-utils"
  }
}
