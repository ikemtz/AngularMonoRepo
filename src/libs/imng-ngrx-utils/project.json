{
  "name": "imng-ngrx-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/imng-ngrx-utils/src",
  "prefix": "imng",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nx/angular:package",
      "outputs": ["{workspaceRoot}/dist/libs/imng-ngrx-utils"],
      "dependsOn": [
        {
          "target": "build",
          "dependencies": true
        }
      ],
      "options": {
        "project": "libs/imng-ngrx-utils/ng-package.json"
      },
      "configurations": {
        "production": {
          "tsConfig": "libs/imng-ngrx-utils/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/imng-ngrx-utils/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/imng-ngrx-utils"],
      "options": {
        "jestConfig": "libs/imng-ngrx-utils/jest.config.ts"
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint"
    }
  }
}
