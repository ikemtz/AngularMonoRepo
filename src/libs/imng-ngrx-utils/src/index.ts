export { Payload, createPayloadAction } from './lib/payload';
export { isTruthy, isFalsy } from './lib/isTruthy';
export { findAndModify } from './lib/find-and-modify';
export { findAndMerge } from './lib/find-and-merge';
export { matchAndSpread } from './lib/match-and-spread';
export { Subscriptions } from './lib/subscriptions';
export { Subscribable } from './lib/subscribable';
export { idType } from './lib/id-type';
