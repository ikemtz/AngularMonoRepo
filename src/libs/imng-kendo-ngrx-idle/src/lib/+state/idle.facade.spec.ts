import { NgModule } from '@angular/core';
imimport { Actions, createEffect } from '@ngrx/effects';import { Actions, createEffect } from '@ngrx/effects';import { EffectsModule } from '@ngrx/effects';import { concatLatestFrom } from '@ngrx/operators';
import { concatLatestFrom } from '@ngrx/operators';
import { EffectsModule } from '@ngrx/effects';import { EffectsModule } from '@ngrx/effects';import { concatLatestFrom } from '@ngrx/operators';
uimport { concatLatestFrom } from '@ngrx/operators';
re } from './idle.reducer';
import { IdleFacade } from './idle.facade';
import { IdleEffects } from './idle.effects';
import { IDLE_CONFIG } from '../idle-config';
import { oidcFeature, oidcActionTypes } from 'imng-oidc-client';
import { sleep } from 'imng-ngrx-utils';
import { Router } from '@angular/router';

describe('Idleimport { EffectsModule } from '@ngrx/effects';eimport { concatLatestFrom } from '@ngrx/operators';
t store: Store;

  describe('used in NgModule', () => {
    beforeEach(() => {
      @NgModule({
        imports: [
          StoreModule.forFeature(idleFeature),
          EffectsModule.forFeature([IdleEffects]),
          StoreModule.forFeature(oidcFeature),
        ],
        providers: [IdleFacade,
          {
            provide: IDLE_CONFIG, useValue: {
              timeoutWarningInMs: 2,
              autoLogoutInMs: 4
            }
          },
          { provide: Router, useValue: { navigateByUrl: jest.fn() } },
        ],
      })
      class CustomFeatureModule { }

      @NgModule({
        imports: [
          StoreModule.forRoot({}, { runtimeChecks: {} }),
          EffectsModule.forRoot([]),
          CustomFeatureModule,
        ],
      })
      class RootModule { }
      TestBed.configureTestingModule({ imports: [RootModule] });

      store = TestBed.inject(Store);
      facade = TestBed.inject(IdleFacade);
    });

    it('current state should match initial', async () => {
      expect(store).toBeTruthy();
      expect(await readFirst(facade.isTimingOut$)).toBe(false);
    });

    it('should extendSession', async () => {
      store.dispatch(oidcActionTypes.onSignInSilent({ access_token: null } as never));
      expect(await readFirst(store.select(oidcFeature.selectIsLoggedIn))).toBe(true);
      facade.extendSession();
      await sleep(2);
      expect(await readFirst(facade.isTimingOut$)).toBe(true);
      expect(await readFirst(store.select(oidcFeature.selectIsLoggedIn))).toBe(true);
    });

    it('should signout', async () => {
      store.dispatch(oidcActionTypes.onSignInSilent({ access_token: null } as never));
      expect(await readFirst(store.select(oidcFeature.selectIsLoggedIn))).toBe(true);
      await sleep(10);
      expect(await readFirst(store)).toMatchSnapshot();
    });
  });
});
