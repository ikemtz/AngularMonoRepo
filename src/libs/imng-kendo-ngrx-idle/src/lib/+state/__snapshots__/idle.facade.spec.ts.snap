// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdleFacade used in NgModule should signout 1`] = `
{
  "idle": {
    "isTimingOut": false,
    "timeoutSpanInMs": undefined,
  },
  "oidc": {
    "audiences": undefined,
    "errors": {
      "httpError": undefined,
      "signInError": undefined,
      "silentRenewError": undefined,
    },
    "identity": undefined,
    "isExpired": false,
    "isExpiring": false,
    "isLoading": true,
    "isLoggedIn": false,
    "permissions": undefined,
    "userMetadata": null,
  },
}
`;
